frontstage/views/secure_messaging.py

```diff
@@ -17,6 +17,91 @@
+        ru_id = get_party_ru_id(party_id)
+        if ru_id is None:
+            return redirect(url_for('error_bp.default_error_page'))
+        survey_name = get_survey_id(party_id)
+        if survey_name is None:
+            return redirect(url_for('error_bp.default_error_page'))
+
+        # Create message object to send to secure-message service
+        data = {'msg_to': ['BRES'],
+                'msg_from': session['party_id'],
+                'subject': request.form['secure-message-subject'],
+                'body': request.form['secure-message-body'],
+                'thread_id': request.form['secure-message-thread-id'],
+                'collection_case': collection_case,
+                'ru_id': ru_id,
+                'survey': survey_name}
+
+        # Set msg_id if message already saved as draft
+        if "msg_id" in request.form:
+            data["msg_id"] = request.form['msg_id']
+            loggerb = loggerb.bind(message_id=data["msg_id"])
+
+        if request.form['submit'] == 'Send':
+            return send_message(data, loggerb)
+
+        elif request.form['submit'] == 'Save draft':
+            return save_draft(data, loggerb)
+
+    return render_template('secure-messages/secure-messages-view.html', _theme='default', message={})
+
+
+@secure_message_bp.route('/<label>/<message_id>', methods=['GET'])
+@jwt_authorization(request)
+def message_get(session, label, message_id):
+    """Get message"""
+    party_id = session['party_id']
+    loggerb = logger.bind(message_id=message_id, party_id=party_id, label=label)
+    message = get_message(message_id, label, loggerb)
+
+    # If message is a draft try and get the last message from its thread
+    if label == 'DRAFT':
+        draft = message
+        thread_id = draft.get('thread_id')
+        if thread_id != draft['msg_id']:
+            message = get_thread_message(thread_id, loggerb, party_id)
+        else:
+            message = None
+    else:
+        draft = {}
+
+    if label == 'INBOX':
+        remove_unread_label(message_id, loggerb)
+
+    return render_template('secure-messages/secure-messages-view.html',
+                           _theme='default',
+                           message=message,
+                           draft=draft,
+                           label=label)
+
+
+@secure_message_bp.route('/messages/', methods=['GET'])
+@secure_message_bp.route('/messages/<label>', methods=['GET'])
+@jwt_authorization(request)
+def messages_get(session, label="INBOX"):
+    """Gets users messages"""
+    party_id = session['party_id']
+    loggerb = logger.bind(party_id=party_id, label=label)
+    messages = get_messages(label, loggerb)
+    unread_msg_total = get_unread_message_total(loggerb)
+    return render_template('secure-messages/secure-messages.html', _theme='default', messages=messages['messages'],
+                           links=messages['_links'], label=label, total=unread_msg_total['total'])
+
+
 def get_party_ru_id(party_id):
```

Comment:
Could you double check the log levels in this function. Don't think it should be an error for a 404

Reply:
Are we sure that this shouldnt be an error? If the user has an account and is sending a secure message they should have all of these things, if they don't something has gone wrong